.goreleaser-darwin.yaml

```diff
@@ -11,8 +11,6 @@ builds:
       - -X github.com/flashbots/mev-boost/config.Version={{.Version}}
     env:
       - CGO_ENABLED=1
```

Comment:
I think we should hardcode this to `CGO_ENABLED=0` now. This would further simplify things. With cgo disabled, go builds are static by default. We'd also be able to 

* remove `osusergo` and `netgo` build tags,
* remove `-extldflags=-static` configuration,
* remove `CC=<compiler>` environment variables,
* remove the `build-static` makefile rule,
* remove the `EXTRA_ENV` makefile variable, and
* join ` GO_BUILD_LDFLAGS` into `GO_BUILD_FLAGS`.

Reply:
Great suggestion! Added a commit that does part of the cleanup. 

Would be amazing if you could combine the `gorelease-*.yml` files in another PR 🥺

Reply:
Nice 👍 yes, I can clean up the releases in another PR.